samples/bot-calling-meeting/csharp/Source/CallingBotSample/Services/TeamsRecordingService/TeamsRecordingService.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace CallingBotSample.Services.TeamsRecordingService
+{
+    public class TeamsRecordingService : ITeamsRecordingService
+    {
+        private readonly IWebHostEnvironment environment;
+        private readonly HttpClient httpClient;
+        private readonly ILogger<TeamsRecordingService> logger;
+        private const string DOWNLOAD_DIRECTORY = "temp";
+
+        public TeamsRecordingService(IWebHostEnvironment environment, HttpClient httpClient, ILogger<TeamsRecordingService> logger)
+        {
+            this.environment = environment;
+            this.httpClient = httpClient;
+            this.logger = logger;
+        }
+
+        public async Task<string> DownloadRecording(string recordingUrl, string accessToken)
+        {
+            var fileName = $"{Guid.NewGuid()}.wav";
+            var downloadDirectory = Path.Combine(environment.WebRootPath, DOWNLOAD_DIRECTORY);
+            var downloadPath = $"{downloadDirectory}/{fileName}";
+
+            var content = await GetUrlContent(recordingUrl, accessToken);
+
+            if (content == null)
+            {
+                throw new Exception("Unable to download file");
```

Comment:
What does the user hear if the bot runs into an error like this?

Reply:
They'll hear nothing unless there is a separate prompt that is playing. This just throws an exception that is handled globally and the prompt is never played to the user.

Reply:
I see. Not blocking this PR but should we play something so the user knows that there was an error processing their request?